test/blackbox-tests/test-cases/describe/describe-pp/describe-pp.t/run.t

```diff
@@ -11,4 +11,4 @@ We can also show the original source if it is not preprocessed
 We also make sure that the dump file is not present
 
   $ dune_cmd exists profile.dump
-  true
+  false
```

Comment:
I think this needs to be updated, since the cwd for running the compiler is different the compiler dumps `profile.dump` in some other place (namely `Path.build (Super_context.context sctx).build_dir`) so presumably something like

```
$ dune_cmd exists _build/default/profile.dump
false
```

Otherwise the `unlink` can still be not triggered and at the same time passing the test just fine.

Reply:
The original bug was about the dump appearing in the source tree. I'll add a test for whatever is happening in the _build dir.

Reply:
Hmm actually I confused myself here. There are two kinds of .dump files. We have `profile.dump` which is what we were seeing in the source tree and we have the individual module dumps. We can see that the module dumps are being unlinked, but in the end it doesn't actually matter since it all happens in the build directory. Therefore I think this is fine as it is.

Reply:
Should we then even try unlinking them if they're in the `_build` directory? If it doesn't matter, then maybe there is not much point in wasting IO and time. What is our policy for this @rgrinberg?

Reply:
Tbh, it's not even a good idea to write them in the _build directory either. Ideally, we'd copy the input into some temp dir and then run it there. Yes, we should clean up after ourselves if we are going to pollute the build dir.